src/main/java/gui/logic/BusinessLogic.java

```diff
@@ -0,0 +1,118 @@
+        try {
+            model = generator.generate();
+        } catch (NoSuchMethodException e) {
+            System.err.println("Problem during generating scenario");
+            e.printStackTrace();
+        } catch (IllegalAccessException e) {
+            System.err.println("Problem during generating scenario");
+            e.printStackTrace();
+        } catch (InvocationTargetException e) {
+            System.err.println("Problem during generating scenario");
+            e.printStackTrace();
+        }
+        DecisionGenerator decisionGenerator = new DecisionGenerator(factory, baseIRI);
+        decisionGenerator.generate(model);
+        return model;
+    }
+
+    public static Map<Decision, Set<Actor>> getCollidedEntities(IConsequenceContainer consequenceContainer, MyFactory factory, Model scenarioModel) {
+        CollisionConsequencePredictor collisionConsequencePredictor =
+                new CollisionConsequencePredictor(consequenceContainer, factory, scenarioModel);
+        SimulatorEngine simulatorEngine = new SimulatorEngine(scenarioModel, collisionConsequencePredictor);
+        Set leftLanes = scenarioModel.getLanes().get(Model.Side.LEFT).entrySet();
+        int lastLeftLane = leftLanes.size();
+        Set rightLanes = scenarioModel.getLanes().get(Model.Side.RIGHT).entrySet();
+        int lastRightLane = rightLanes.size();
+        Map<Decision, Set<Actor>> collidedEntities = simulatorEngine.simulateAll(lastLeftLane, lastRightLane);
+
+        // pytanie czy może to być w tym miejscu, bo umieszczenie tego później miałoby sens,
+        // jeśli korzytalibyśmy z MoralDilemmaDetector
+        try {
+            factory.saveOwlOntology();
+        } catch (OWLOntologyStorageException e) {
+            System.err.println("Problem during saving ontology");
+            e.printStackTrace();
+        }
+        return collidedEntities;
+    }
+
+    // optymalnie jechać cały nie wykonywać manewrów,
+    // jeśli konieczny to preferowane są w prawo ze względu na ruch prawostronny
+    public static String getOptimumDecision(Map<String, Integer> decisionCosts) {
+        ArrayList<String> decisionsWithMinimalCost = new ArrayList<>();
+        int currentMinimum = Integer.MAX_VALUE;
+        for (String decisionName : decisionCosts.keySet()) {
+            Integer cost = decisionCosts.get(decisionName);
+            if (cost < currentMinimum) {
+                currentMinimum = cost;
+                decisionsWithMinimalCost = new ArrayList<>();
+                decisionsWithMinimalCost.add(decisionName);
+            }
+            if (cost == currentMinimum) {
+                decisionsWithMinimalCost.add(decisionName);
+            }
+        }
+        if (decisionsWithMinimalCost.size() == 1) {
+            return decisionsWithMinimalCost.get(0);
+        }
+        if (decisionsWithMinimalCost.contains("follow")) {
+            return "follow";
+        }
+
+        for (int i = 1; i < 10; i++) {
```

Comment:
Magic number 10 zakładający, że nie istnieją drogi o większej liczbie pasów niż 10

Reply:
Może przeiterować po tych decyzjach i regexem to sprawdzić?

Reply:
O to jest spoko, nie pomyślałem o tym :)